### ISSUE TITLE
Downstream errors on 32bit

### ISSUE THREAD

**Issue by devmotion** [LilithHafner/AliasTables.jl#48]
Downstream tests in Turing on 32bit recently started to fail due to AliasTables (or its integration in Distributions): https://github.com/TuringLang/Turing.jl/actions/runs/8868873134/job/24348938461#step:6:588
```julia
  ArgumentError: Lookup table longer than length(probablity_alias)
  Stacktrace:
    [1] _lookup_alias_table!(probability_alias::Vector{Tuple{UInt64, Int32}}, weights::AliasTables.MallocArrays.MallocArray{UInt64, 1}, mtz::Int32)
      @ AliasTables ~/.julia/packages/AliasTables/wd9Qk/src/AliasTables.jl:182
    [2] _alias_table!(probability_alias::Vector{Tuple{UInt64, Int32}}, weights::AliasTables.MallocArrays.MallocArray{UInt64, 1})
      @ AliasTables ~/.julia/packages/AliasTables/wd9Qk/src/AliasTables.jl:248
    [3] set_weights!(at::AliasTables.AliasTable{UInt64, Int32}, weights::Vector{Float64}; _normalize::Bool)
      @ AliasTables ~/.julia/packages/AliasTables/wd9Qk/src/AliasTables.jl:159
    [4] _
      @ ~/.julia/packages/AliasTables/wd9Qk/src/AliasTables.jl:115 [inlined]
    [5] #AliasTable#3
      @ ~/.julia/packages/AliasTables/wd9Qk/src/AliasTables.jl:119 [inlined]
    [6] AliasTable
      @ ~/.julia/packages/AliasTables/wd9Qk/src/AliasTables.jl:119 [inlined]
    [7] AliasTable
      @ ~/.julia/packages/Distributions/fgrZq/src/samplers/aliastable.jl:3 [inlined]
    [8] sampler
      @ ~/.julia/packages/Distributions/fgrZq/src/univariate/discrete/categorical.jl:118 [inlined]
    [9] rand(rng::StableRNGs.LehmerRNG, s::Categorical{Float64, Vector{Float64}}, dims::Tuple{Int32})
      @ Distributions ~/.julia/packages/Distributions/fgrZq/src/genericrand.jl:35
   [10] rand(::StableRNGs.LehmerRNG, ::Categorical{Float64, Vector{Float64}}, ::Int32)
      @ Distributions ~/.julia/packages/Distributions/fgrZq/src/genericrand.jl:24
...
```

My initial guess is that some part of AliasTables implicitly assumes that `Int = Int64`.

**Comment by LilithHafner** [comment]
Thank you for reporting this.

**Comment by LilithHafner** [comment]
This issue does not reproduce on:
- Julia 1.10.2, 32-bit Linux, AliasTables 1.1.0
- Julia 1.10.2, 64-bit Aarch, AliasTables 1.1.0
- Julia 1.6.7 32-bit Linux, AliasTables 1.0.0

Does reproduce on
- Julia 1.6.7 32-bit Linux, AliasTables 1.1.0

```julia
julia> AliasTable([0.5, 0.5])
ERROR: UndefVarError: AliasTable not defined
Stacktrace:
 [1] top-level scope
   @ REPL[4]:1

julia> using AliasTables

julia> AliasTable([0.5, 0.5])
ERROR: ArgumentError: Lookup table longer than length(probablity_alias)
Stacktrace:
 [1] _lookup_alias_table!(probability_alias::Vector{Tuple{UInt64, Int32}}, weights::AliasTables.MallocArrays.MallocArray{UInt64, 1}, mtz::Int32)
   @ AliasTables ~/.julia/packages/AliasTables/wd9Qk/src/AliasTables.jl:182
 [2] _alias_table!(probability_alias::Vector{Tuple{UInt64, Int32}}, weights::AliasTables.MallocArrays.MallocArray{UInt64, 1})
   @ AliasTables ~/.julia/packages/AliasTables/wd9Qk/src/AliasTables.jl:248
 [3] set_weights!(at::AliasTable{UInt64, Int32}, weights::Vector{Float64}; _normalize::Bool)
   @ AliasTables ~/.julia/packages/AliasTables/wd9Qk/src/AliasTables.jl:159
 [4] _
   @ ~/.julia/packages/AliasTables/wd9Qk/src/AliasTables.jl:115 [inlined]
 [5] #AliasTable#3
   @ ~/.julia/packages/AliasTables/wd9Qk/src/AliasTables.jl:119 [inlined]
 [6] AliasTable(weights::Vector{Float64})
   @ AliasTables ~/.julia/packages/AliasTables/wd9Qk/src/AliasTables.jl:119
 [7] top-level scope
   @ REPL[6]:1

julia> versioninfo()
Julia Version 1.6.7
Commit 3b76b25b64 (2022-07-19 15:11 UTC)
Platform Info:
  OS: Linux (i686-pc-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU           E5520  @ 2.27GHz
  WORD_SIZE: 32
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, nehalem)

(jl_41HCSR) pkg> st
      Status `/tmp/jl_41HCSR/Project.toml`
  [66dad0bd] AliasTables v1.1.0
```

**Comment by LilithHafner** [comment]
This is the problem:
```julia
top_set_bit(x::Int) = 64 - leading_zeros(x)
```

**Comment by LilithHafner** [comment]
It will effect most usages of AliasTables on all 32-bit systems on Julia 1.8 and below on AliasTables version 1.1.0 (and no other versions of AliasTables).

**Comment by LilithHafner** [comment]
Fixed on the new 1.1.1 release! Sorry for the disruption.
